src/olympia/amo/decorators.py

```diff
@@ -181,3 +181,48 @@ def wrapper(request, *args, **kw):
         return response
 
     return wrapper
+
+
+def api_authentication(f):
+    """Allows API authentication to be used by this django view. Standard auth will
+    already have been attempted by this point so api auth will only be tried for
+    anonymous (unauthenticated) requests."""
+
+    from rest_framework import exceptions as drf_exceptions
+    from rest_framework.settings import api_settings
+    from rest_framework.views import exception_handler as drf_exception_handler
+
+    from olympia.api.authentication import (
+        get_authorization_header,
+        SessionIDAuthentication,
+        JWTKeyAuthentication,
+    )
+
+    @functools.wraps(f)
+    def wrapper(request, *args, **kw):
+        if request.user.is_anonymous and get_authorization_header(request):
+            # if user isn't authenticated with standard auth, try the API auth methods
+            try:
+                for api_auth in (SessionIDAuthentication, JWTKeyAuthentication):
+                    api_auth = api_auth()
+                    result = api_auth.authenticate(request)
+                    if result:
+                        request.user, _ = result
+                        break
+            except drf_exceptions.AuthenticationFailed as exc:
+                # We have to set some props DRF would usually set in the APIView
```

Comment:
Do we need that/care ?

Reply:
the comment or setting those properties?

Reply:
setting those properties - given that the underlying view will _not_ be an API view

Reply:
it's because we're explicitly returning an API style error response rather than just re-raising and returning a normal django view response - I figured that was more "correct" as any client ideally doesn't want to have to handle an entirely different auth error response just for that view.